src/app/shared/data/allocations.service.ts

```diff
@@ -33,16 +33,37 @@ export class AllocationsService {
     return this.allocationsSubject$.getValue();
   }
 
-  addStudentToProject(studentId: string, projectId: string): void {
+  get allocations$(): Observable<Allocation[]> {
+    return this.allocationsSubject$.asObservable();
+  }
+
+  moveStudentToProject(studentId: string, projectId: string): void {
     this.removeStudentFromProjects(studentId);
 
     const allocations = this.getAllocations();
     const allocation = allocations.find(a => a.projectId === projectId);
 
     if (allocation) {
       allocation.students.push(studentId);
+    } else {
+      allocations.push({ projectId, students: [studentId] });
+    }
+    this.setAllocations(allocations);
+  }
+
+  moveStudentToProjectAtInset(studentId: string, projectId: string, siblingId?: string): void {
+    this.removeStudentFromProjects(studentId);
+
+    const allocations = this.getAllocations();
+    const allocation = allocations.find(a => a.projectId === projectId);
+
+    if (allocation && siblingId) {
+      const index = allocation.students.indexOf(siblingId);
```

Comment:
I would also create a descriptive function for that part. That makes it easier to understand what you're doing here